dts/bindings/timer/nordic,nrf-timer.yaml

```diff
@@ -32,3 +32,15 @@ properties:
   zli:
     type: boolean
     description: Enable event handler in the ZLI (Zero latency interrupt) context
+
+  instance:
+    type: string
+    required: true
+    description: |
+      The TIMER instance id as used in the MDK and documentation. For example for TIMER2:
+
+        instance = "2";
+
+      And for TIMER00:
+
+        instance = "00";
```

Comment:
We have previously rejected similar work-arounds for other SoCs. Could we find another way around this?

Reply:
I won't nack here, but I agree that this is a DT usage violation we have rejected multiple times. If HALs do not match DT expectations, they should be modified to do so, use other layers (eg `nrf_`), or not use them at all.

Reply:
This change is part of one of the commits fixing https://github.com/zephyrproject-rtos/zephyr/issues/69216 . Here is an alternative fix for the same issue: #69966. If it gets merged, I will drop these changes from my PR.